lib/charms/layer/azure.py

```diff
@@ -174,7 +174,12 @@ def send_additional_metadata(request):
     available from the metadata server.
     """
     run_config = hookenv.config() or {}
-    res_grp = _azure("group", "show", "--name", request.resource_group)
+    res_grp_args = ["group", "show", "--name", request.resource_group]
+    if hasattr(request, 'subscription_id'):
+        for pos, arg in enumerate(["--subscription", request.subscription_id]):
+            res_grp_args.insert(pos + 2, arg)
+
+    res_grp = _azure(*res_grp_args)
```

Comment:
i mean, if order doesn't matter... these commands are the same right?
```bash
azure group show --subscription <id> --name <group>
```

```bash
azure group show --name <group> --subscription <id> 
```

Please ignore my previous suggestion in favor of this one:
```suggestion
    res_grp_args = ["group", "show", "--name", request.resource_group]
    if hasattr(request, 'subscription_id'):
        res_grp_args += ["--subscription", request.subscription_id]
    res_grp = _azure(*res_grp_args)
```

Reply:
added this